web/client/components/theme/ContextTheme.jsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+
+/**
+ * Theme component provider for the context pages
+ * @memberof components.theme
+ * @name ContextTheme
+ * @prop {array} theme a list of theme configurations to apply in a specific context
+ * @example
+ * // add a css link tag in the context page
+ * <ContextTheme
+ *  theme={[{
+ *      id: 'dark',
+ *      type: 'link',
+ *      href: 'dist/themes/dark.css'
+ *  }]}/>
```

Comment:
```suggestion
 * <ContextTheme
 *  theme={[{
 *      id: 'dark',
 *      type: 'link',
 *      href: 'dist/themes/dark.css'
 *  },
 *  {
 *     id: 'custom-variables',
 *     type: 'css-variables',
 *     variables: {
 *         primary: '#ff0000'
 *    }
 *   }]}/>
```

Reply:
variable are not currently supported. I think we should improve the doc only when we add them